src/metabase/driver/sql/query_processor.clj

```diff
@@ -181,18 +181,29 @@
   [_ field]
   (apply hsql/qualify (field/qualified-name-components field)))
 
-(defmulti ^String field->alias
-  "Return the string alias that should be used to for `field`, an instance of the Field model, i.e. in an `AS` clause.
-  The default implementation calls `:name`, which returns the *unqualified* name of the Field.
+(defmulti ^String escape-alias
+  "Return the String that should be emitted in the query for the given `alias-name`, which will follow the `AS` clause.
+  This is to allow for escaping names that particular databases may not allow as aliases for custom expressions
+  or fields (even when quoted).
 
-  Return `nil` to prevent `field` from being aliased."
-  {:arglists '([driver field])}
```

Comment:
It might be a good idea to just leave this method unused for the time being and mark it as deprecated just so it's not a breaking change for 3rd-party drivers in 41. Normally we've been marking something as deprecated and removing it 3 versions later.

Not sure anybody else ever actually implemented this method but since it's not `^:private` we can't be sure